File: cache/cache-annotation-processor/src/test/java/io/koraframework/cache/annotation/processor/testcache/DummyCache21.java
package io.koraframework.cache.annotation.processor.testcache;

import io.koraframework.cache.annotation.Cache;
import io.koraframework.cache.caffeine.CaffeineCache;

import java.math.BigDecimal;

@Cache("dummy21")
public interface DummyCache21 extends CaffeineCache<DummyCache21.Key, String> {

    record Key(String k1, BigDecimal k2) {}
}

File: cache/cache-symbol-processor/src/test/kotlin/io/koraframework/cache/symbol/processor/testcache/DummyCache21.kt
package io.koraframework.cache.symbol.processor.testcache

import io.koraframework.cache.annotation.Cache
import io.koraframework.cache.caffeine.CaffeineCache
import java.math.BigDecimal

@Cache("dummy21")
interface DummyCache21 : CaffeineCache<DummyCache21.Key, String> {

    data class Key(val k1: String, val k2: BigDecimal)
}
